lib/ExportsStorage.js

```diff
@@ -0,0 +1,241 @@
+import path from 'path';
+
+import sqlite3 from 'sqlite3';
+
+import lastUpdate from './lastUpdate';
+
+const MAX_CHUNK_SIZE = 100;
+
+function normalizedExportName(string) {
+  return string.toLowerCase().replace(/[-_.]/g, '');
+}
+
+function inParam(sql, values) {
+  // https://github.com/mapbox/node-sqlite3/issues/721
+  return sql.replace('?#', values.map(() => '?').join(','));
+}
+
+function arrayToChunks(array, chunkSize) {
+  const chunks = [];
+  for (let i = 0; i < array.length; i += chunkSize) {
+    chunks.push(array.slice(i, i + chunkSize));
+  }
+  return chunks;
+}
+
+export default class ExportsStorage {
+  init(dbFilename) {
+    return new Promise((resolve, reject) => {
+      this.db = new sqlite3.Database(dbFilename);
+      this.db.run(`
+        CREATE TABLE IF NOT EXISTS exports (
+          name VARCHAR(100),
+          isDefault INTEGER,
+          path TEXT
+        )
+      `, (err) => {
+        if (err) {
+          reject(err);
+          return;
+        }
+
+        this.db.run(`
+          CREATE TABLE IF NOT EXISTS mtimes (
+            path TEXT,
+            mtime NUMERIC
+          )
+        `, (err) => {
+          if (err) {
+            reject(err);
+            return;
+          }
+
+          resolve();
+        });
+      });
+    });
+  }
+
+  close() {
+    return new Promise((resolve, reject) => {
+      this.db.close((err) => {
+        if (err) {
+          reject(err);
+          return;
+        }
+        resolve();
+      });
+    });
+  }
+
+  needsUpdate(files) {
```

Comment:
Yeah, it's hard to step away from the object-oriented past self. I'll think about this some more, at least this file doesn't have that much state (`ModuleFinder` is worse).